src/firefly/js/ui/tap/TapUtil.js

```diff
@@ -218,13 +218,24 @@ export function getColumnAttribute(columnsModel, colName, attrName) {
 
 const hasElements= (a) => Boolean(isArray(a) && a?.length);
 
+function mergeAdditionalServices(tapServices, additional) {
+    if (!hasElements(additional)) return tapServices;
+
+    const modifiedOriginal= tapServices.map( (t) => {
+        const match= additional.find( (a) => a.label===t.label);
+        return match ? {...t,...match} : t;
+    });
+    const trulyAdditional= additional.filter( (a) => tapServices.filter( (t) => t.label===a.label).length===0 );
+
+    return [...trulyAdditional,...modifiedOriginal];
+}
+
 export function getTapServices(webApiUserAddedService) {
-    const tapServices = getAppOptions()?.tap?.services;
-    const additionalServices = getAppOptions()?.tap?.additional?.services;
-    const retVal= hasElements(tapServices) ? [...tapServices] : [...TAP_SERVICES_FALLBACK];
-    hasElements(additionalServices) && retVal.unshift(...additionalServices);
-    webApiUserAddedService && retVal.push(webApiUserAddedService);
-    return retVal;
+    const {tap} = getAppOptions();
+    const startingTapServices= hasElements(tap?.services) ? [...tap.services] : [...TAP_SERVICES_FALLBACK];
+    const mergedServices= mergeAdditionalServices(startingTapServices,tap?.additional?.services);
```

Comment:
For the purpose of adding additional services to the top of the list, this works fine.  However, a more elaborate scheme is needed to truly manage the list, like insert and remove.